### DISCUSSION TITLE
Preset different variants to a model at the same time.

### DISCUSSION THREAD

**Discussion by chesstrian** [google/model-viewer#4667] [Q&A]
Hi guys,

I have a question about setting more than one variant to a model in different material:

Given a model with different material variants like the one shown in https://github.com/google/model-viewer/discussions/3822#discussioncomment-3712516, let's say I need to set it up with different variants than the default ones, with javascript it'd be:

```js
function setVariants() {
  modelViewer.variantName = 'Dash Pink';
  modelViewer.variantName = 'Natural Olive Ash';
}
```

However when the function is called, only the second variant is applied:

<img width="1440" alt="image" src="https://github.com/google/model-viewer/assets/3923412/b2d7790e-42d9-41db-9e0c-cc84d5234116">

In this case it is expected to get:

<img width="1440" alt="image" src="https://github.com/google/model-viewer/assets/3923412/d8ae093a-997f-4504-81f7-08ecac8ed156">

Is it possible to get both variants applied at the same time?.

Thanks in advance.

**Comment by elalish** [comment]
No, the way the glTF variants extension works is that a named variant is a complete choice of materials for all meshes. Of course if you have a bunch of permutations, this leads to an explosion of named variants. In that case you're better off skipping variants and using our `model` API to switch materials directly.
